code/modules/admin/verbs/SDQL2/README.md

```diff
@@ -0,0 +1,177 @@
+```sql
+SELECT /mob WHERE z == 4
+```
+
+What if you want to select every mob on even numbered z-levels?
+```sql
+SELECT /mob WHERE z % 2 == 0
+```
+
+Can you see where this is going? You can select objects with an arbitrary expression. These expressions can also do variable access and proc calls (yes, both on-object and globals!). Keep reading!
+
+### In, where, and *
+Okay. What if you want to get every machine in the `SSmachine` process list? Looping through world is kinda slow.
+
+```sql
+SELECT * IN SSmachines.processing
+```
+
+Here `*` as type functions as a wildcard. We know everything in the global `SSmachines.processing` list is a machine.
+
+You can specify `IN <expression>` to return a list to operate on. This can be any list that you can wizard together from global variables and global proc calls. Every variable/proc name in the `IN` block is global. It can also be a single object, in which case the object is wrapped in a list for you. So yeah SDQL is unironically better than VV for complex single-object operations.
+
+You can of course combine these.
+```sql
+SELECT * IN SSmachines.processing WHERE z == 4
+SELECT * IN SSmachines.processing WHERE stat & 2 // (2 is NOPOWER, cannot use defines from SDQL. Sorry!)
+SELECT * IN SSmachines.processing WHERE stat & 2 && z == 4
+```
+
+The possibilities are endless (just don't crash the server, okay?).
+
+### Map and selector arrays
+Oh it gets better.
+
+You can use `MAP <expression>` to run some code per object and use the result. For example:
+
+```sql
+SELECT /obj/machinery/power/smes MAP [charge / capacity * 100, RCon_tag, src]
+```
+
+This will give you a list of all the APCs, their charge, AND RCon tag. Useful eh?
+
+`[]` being a list here. Yeah you can write out lists directly without > lol lists in VV. Color matrix
+shenanigans inbound.
+
+After the `MAP` segment is executed, the rest of the query executes as if it's THAT object you just made
+(here the list).
+Yeah, by the way, you can chain these `MAP` / `WHERE` things FOREVER!
+
+```sql
+SELECT /mob WHERE client MAP client WHERE holder MAP holder
+```
+
+You can also generate a new list on the fly using a selector array. `@[]` will generate a list of objects based off the selector provided.
+
+```sql
+SELECT /mob/living IN (@[/area/crew_quarters/bar MAP contents])[1]
+```
+
+### Delete
+
+What if some dumbass admin spawned a bajillion spiders and you need to kill them all?
+Oh yeah you'd rather not delete all the spiders in maintenance. Only that one room the spiders were
+spawned in.
+
+```sql
+DELETE /mob/living/carbon/superior_animal/giant_spider WHERE loc.loc == marked
+```
+
+Here I used VV to mark the area they were in, and since `loc.loc == area`, voila.
+Only the spiders in a specific area are gone.
+
+Or you know if you want to catch spiders that crawled into lockers too (how even?)
+
+```sql
+DELETE /mob/living/carbon/superior_animal/giant_spider WHERE global.get_area(src) == marked

```

Comment:
May want to make these the actual types we have here, `/mob/living/simple_animal/hostile/poison/giant_spider`